a10_octavia/network/drivers/neutron/a10_octavia_neutron.py

```diff
@@ -12,697 +12,189 @@
+            pass
         except Exception:
-            message = _('Error plugging amphora (compute_id: {compute_id}) '
-                        'into network {network_id}.').format(
-                            compute_id=compute_id,
-                            network_id=network_id)
+            message = "Error creating port details: ip {ip} "
+            "subnet {subnet_id}".format(ip=ip, subnet_id=subnet.id)
             LOG.exception(message)
-            raise base.PlugNetworkException(message)
-
-        return self._nova_interface_to_octavia_interface(compute_id, interface)
-
-    def unplug_network(self, compute_id, network_id, ip_address=None):
-        interfaces = self.get_plugged_networks(compute_id)
-        if not interfaces:
-            msg = ('Amphora with compute id {compute_id} does not have any '
-                   'plugged networks').format(compute_id=compute_id)
-            raise base.NetworkNotFound(msg)
-
-        unpluggers = self._get_interfaces_to_unplug(interfaces, network_id,
-                                                    ip_address=ip_address)
-        for index, unplugger in enumerate(unpluggers):
-            self.compute.detach_port(
-                compute_id=compute_id, port_id=unplugger.port_id)
-
-    def update_vip(self, load_balancer, for_delete=False):
-        sec_grp = self._get_lb_security_group(load_balancer.id)
-        if sec_grp:
-            self._update_security_group_rules(load_balancer, sec_grp.get('id'))
-        elif not for_delete:
-            raise exceptions.MissingVIPSecurityGroup(lb_id=load_balancer.id)
-        else:
-            LOG.warning('VIP security group missing when updating the VIP for '
-                        'delete on load balancer: {lb_id}. Skipping update '
-                        'because this is for delete.'.format(
-                            lb_id=load_balancer.id))
-
-    def failover_preparation(self, amphora):
-        if self.dns_integration_enabled:
-            self._failover_preparation(amphora)
 
-    def _failover_preparation(self, amphora):
-        interfaces = self.get_plugged_networks(compute_id=amphora.compute_id)
-
-        ports = []
-        for interface_ in interfaces:
-            port = self.get_port(port_id=interface_.port_id)
-            ips = port.fixed_ips
-            lb_network = False
-            for ip in ips:
-                if ip.ip_address == amphora.lb_network_ip:
-                    lb_network = True
-            if not lb_network:
-                ports.append(port)
-
-        for port in ports:
-            try:
-                self.neutron_client.update_port(port.id,
-                                                {'port': {'dns_name': ''}})
-
-            except (neutron_client_exceptions.NotFound,
-                    neutron_client_exceptions.PortNotFoundClient):
-                raise base.PortNotFound()
-
-    def plug_port(self, amphora, port):
+    def get_ve_port_id(self, ip):
```

Comment:
We should just have a default `get_port` that maps it to the datamodel. I believe this utility already exists though in the upstream octavia code base.

Reply:
Yes `get_port` does exist, but I guess, taking this changes from #96 which get port details by Ip address, needs this method. 
Hence changing the method name as `get_port_by_name` and also making it return `network_models.Port(id=port['id'])`

Reply:
Actually this isn't even required. The only reason this was being done is because the `create_ve_port` wasn't returning the port itself. We should never have to get one of the ports we reserve by IP.